src/guiguts/highlight.py

```diff
@@ -0,0 +1,90 @@
+"""Highlight functionality"""
+
+from enum import auto
+
+from guiguts.maintext import maintext
+from guiguts.preferences import preferences, PrefKey
+
+
+class Highlight:
+    """Global highlight settings"""
+
+    TAG_QUOTEMARK = auto()
+
+    # Possible future enhancement:
+    #
+    # GG1 allowed you to set three colors:
+    # - Background color (text areas, text inputs, main editor textarea)
+    # - Button highlight color (hover on buttons, checkboxes, radio selects)
+    # - Scanno/quote highlight color (which would apply here)
+    #
+    # Unclear what we should/will do in GG2 with themes & dark mode support.
+
+    # Must be a definition for each available theme
+    COLORS_QUOTEMARK = {
+        "Light": {"bg": "#a08dfc", "fg": "black"},
+        "Dark": {"bg": "#a08dfc", "fg": "white"},
+        "Default": {"bg": "#a08dfc", "fg": "black"},
```

Comment:
If "fg" is supposed to be white on the `Dark`, it's not working for me (macOS Sonoma). All three of them, I see the same fg and bg colors. I.e. black text on `#a08dfc`.

Reply:
Nevermind (🤦‍♀️). I forgot until I was looking back at it that you have to restart to guarantee that the new theme takes full effect. It's working as advertised.

Reply:
I think the built-in themes in tkinter natively support dark/light mode but I don't see any code in GG that knows the difference when the theme is `Default`.

I found a library that claims to be able to do this detection on Mac, Windows, and Linux, but I haven't tried it.

https://pypi.org/project/darkdetect/

Reply:
> Here `Dark` means the Guiguts theme "Dark" in the preferences - which is not the same thing as using Dark mode in macOS, with `Default` theme in Guiguts. I'll try to figure out how to detect the system dark/light mode when in Default theme.

Yes, I realized that -- and that was what I was seeing, but as my later post mentions, I forgot that for certain things, to guarantee that you're seeing it correctly, you have to restart GG2.

Reply:
Detecting Dark mode from the shell is "easy" although... a bit stupid.

```
❯ defaults read -g AppleInterfaceStyle;echo $?
Dark
0

❯ defaults read -g AppleInterfaceStyle;echo $?
2024-06-21 00:41:52.661 defaults[57537:10085832]
The domain/default pair of (kCFPreferencesAnyApplication, AppleInterfaceStyle) does not exist
1
```

In other words, it doesn't tell you the mode if it's light, you get an error instead. You wouldn't even need to read the output, I guess; the return value of the program tells the story.

I don't propose using this, by the way - although it seems to work, if nothing better emerges...